### ISSUE TITLE
[Feature] Have somewhere to see all starred messages 

### ISSUE THREAD

**Issue by fcoppolani** [RocketChat/Rocket.Chat#10797]
Hi,

As for 0.63.1 version starred messages are only per channel/private message. Some users aske for a way of searching all their starred message without selecting a channel. I guess this option should be available only if global search is enabled.
I would imagine a checkbox 'starred messages' along with the 'global search' allowing a user to search only amongst his starred message : 
![2018-05-17_15h01_39](https://user-images.githubusercontent.com/7274681/40179130-cd9e5508-59e3-11e8-962a-34712367b4d3.png)

could be related to #310 (which has not been totally finished)

Thanks in advance for your help.

Regards,

**Comment by Hudell** [comment]
You can search for `has:star`

**Comment by fcoppolani** [comment]
Thx @Hudell for this, I didn't realize it was possible. A clickable filter would be better but it sure allows me to go back to my users waiting for this.

**Comment by Hudell** [comment]
Yes, it's one of those things that there's no way to know unless you look in the code. We are going to add this information (as well as other search modifiers) to the search screen.

**Comment by saschafoerster** [comment]
I am one of users searching for this and it took my some time to come back to find this thread with a working solution. But still: favorites over all channels are a great way to remember stuff that has to be done later, and if I don't remember the channel, I am lost again and the star-feature is not very useful to it's full extend. https://github.com/RocketChat/Rocket.Chat/issues/11435

**Comment by antkaz** [comment]
Hi RC team. We also have a need for this feature. Maybe make `roomId` as an optional parameter for the `chat.getStarredMessages` endpoint. Then the result will include all stared messages for the user. And for UI add checkbox to display global or local stared messages.

Using global search with the `has: star` filter is neither convenient nor obvious.

**Comment by mslovensky** [comment]
Trying to switch from MatterMost and this may be a deal killer, I like RC but MM's use of flagging is simply and very useful to organize action messages/items.

**Comment by haarbrandt** [comment]
Hey Rocket.Chat. I also want to vote for this feature. With a medium sized company but many channels and chats it's essential to be able to memorize messages to read and/or answer them later. And it's a must have to have an overview about the starred messages on all channels. It's like your mail inbox where you can leave the read messages or label or flag the or whatever.

As other people already mentioned: it's absolutely essential and "may be a deal killer". Look at how Slack is doing it ("Saved items"") and don't fear copying it :-)

**Comment by Kyogre** [comment]
> Using global search with the has:star filter is neither convenient nor obvious.
if somebody don't know how to enable global search: open administration settings -> Search -> Default provider -> enable "Global search" option ([official guide](https://docs.rocket.chat/guides/administration/settings/search)).

**Comment by Kyogre** [comment]
but global search has:star is nearly useless now due to
https://github.com/RocketChat/Rocket.Chat/issues/20574
https://github.com/RocketChat/Rocket.Chat/issues/25032

**Comment by PvUtrix** [comment]
Please add a way to see stared messages globally, across all channels and private chats. 
I need to flag messages somehow, as not to lose them and have them all available in 1 click, like Saved Messages in Telegram.

**Comment by saschafoerster** [comment]
It's also a feature I am really missing. When I star something, it's for coming back to it later, like a reminder. But when I have to check more than 50 channels to find the favorites back, it's rather useless. I would be really happy if I didn't need to remind which channel to search for my favorites. Also remembering some shortcodes for search is not really helpful. A clickable button would be so much better.

**Comment by lezhendima** [comment]
Please add a way to see stared messages globally, across all channels and private chats.
I need to flag messages somehow, as not to lose them and have them all available in 1 click, like Saved Messages in Telegram.

**Comment by thw0rted** [comment]
[This docs page](https://docs.rocket.chat/use-rocket.chat/workspace-administration/settings/search#default-provider) says that Global Search is still in beta.  Is there really no fully-supported way of finding content across all channels and DMs?
